lib/src/functions/map.dart

```diff
@@ -72,19 +71,12 @@ final _merge = BuiltInCallable.overloadedFunction("merge", {
     } else if (args.length == 1) {
       throw SassScriptException("Expected \$args to contain a map.");
     }
-    var keys = args.sublist(0, args.length - 1);
     var map2 = args.last.assertMap("map2");
-    SassMap _mergeCallback(Value oldMap) {
-      // If oldMap is null, this indicates there is no nested map at the key.
-      // If this is the case, set the value at the key to map2, which is akin to
-      // merging map2 into an empty map. This behavior is consistent with that
-      // of map.set().
-      return (oldMap == null)
-          ? map2
-          : SassMap({...oldMap.assertMap().contents, ...map2.contents});
-    }
-
-    return _modify(map1, keys, _mergeCallback);
+    return _modify(map1, args.sublist(0, args.length - 1), (Value oldValue) {
```

Comment:
```suggestion
    return _modify(map1, args.sublist(0, args.length - 1), (oldValue) {
```

This type is inferred from `_modify()`'s signature. That's one of the benefits of writing callbacks inline: you have to specify fewer types manually.